services/server/src/server/services/storageServices/EtherscanVerifyAPIService.ts

```diff
@@ -0,0 +1,586 @@
+import {
+  CompilationLanguage,
+  VerificationExport,
+  VyperSettings,
+} from "@ethereum-sourcify/lib-sourcify";
+import logger from "../../../common/logger";
+import { WStorageService } from "../StorageService";
+import { WStorageIdentifiers } from "./identifiers";
+import { Database } from "../utils/Database";
+import { SourcifyDatabaseService } from "./SourcifyDatabaseService";
+import { sourcifyChainsMap } from "../../../sourcify-chains";
+
+export type EtherscanVerifyAPIIdentifiers =
+  | WStorageIdentifiers.EtherscanVerify
+  | WStorageIdentifiers.BlockscoutVerify
+  | WStorageIdentifiers.RoutescanVerify;
+
+const DEFAULT_ETHERSCAN_CHAINLIST_ENDPOINT =
+  "https://api.etherscan.io/v2/chainlist";
+const DEFAULT_BLOCKSCOUT_CHAINLIST_ENDPOINT =
+  "https://chains.blockscout.com/api/chains";
+const ROUTESCAN_CHAINLIST_ENDPOINTS = [
+  {
+    workspace: "mainnet",
+    endpoint: "https://api.routescan.io/v2/network/mainnet/evm/all/blockchains",
+  },
+  {
+    workspace: "testnet",
+    endpoint: "https://api.routescan.io/v2/network/testnet/evm/all/blockchains",
+  },
+] as const;
+
+type ChainApiUrls = Record<string, string>;
+
+const fetchEtherscanChainApiUrls = async (): Promise<ChainApiUrls> => {
+  const targetEndpoint = DEFAULT_ETHERSCAN_CHAINLIST_ENDPOINT;
+  let response: Response;
+
+  try {
+    response = await fetch(targetEndpoint);
+  } catch (error) {
+    logger.error("Failed to fetch Etherscan chain list", {
+      endpoint: targetEndpoint,
+      error,
+    });
+    throw new Error(
+      `Failed to fetch Etherscan chain list from ${targetEndpoint}`,
+    );
+  }
+
+  if (!response.ok) {
+    const body = await response.text();
+    logger.error("Etherscan chain list returned non-OK response", {
+      endpoint: targetEndpoint,
+      status: response.status,
+      body,
+    });
+    throw new Error(
+      `Failed to load Etherscan chain list (${response.status}): ${body}`,
+    );
+  }
+
+  interface ChainListEntry {
+    chainid?: string;
+    apiurl?: string;
+    status?: number;
+  }
```

Comment:
I think it's better to not mix type definitions with logic and rather define interfaces outside function bodies.